### ISSUE TITLE
Very slow learning speed for RollerBall project 

### ISSUE THREAD

**Issue by Jean-Diddy** [Unity-Technologies/ml-agents#4042]
Hello everyone. I followed the ml-agent RollerBall tutorial and trained the agent on Unity. But when I start the training there are first a warning I do not really understand and then the learning speed is very low compared to what is displayed in the tutorial whereas I put the same configurations for the hyperparameters and add a sequence_length = 64. The graphs which demonstrates this slow learning rate are shown below. The total training time for 50 000 steps is about 15 minutes. To get the same results than in the tutorial I have to put in the scene a dozen of agents...
Do you have any idea which could explain this huge difference ? (Windows 10, AMD Ryzen 5 3 500U, Radeon Vega 8

These are the hyperparameters I put : 

RollerBall:
  trainer: ppo
  batch_size: 10
  beta: 5.0e-3
  buffer_size: 100
  epsilon: 0.2
  hidden_units: 128
  lambd: 0.95
  learning_rate: 3.0e-4
  learning_rate_schedule: linear
  max_steps: 5.0e4
  memory_size: 128
  normalize: false
  num_epoch: 3
  num_layers: 2
  time_horizon: 64
  sequence_length: 64
  summary_freq: 1000
  use_recurrent: false
  reward_signals:
    extrinsic:
      strength: 1.0
      gamma: 0.99

This is one of the warning I get (I do not show you the warnings concerning the GPU) : 

'''2020-05-29 16:37:49.129687: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2'''

The Graphs I obtained : 

![Graphs_RollerBall](https://user-images.githubusercontent.com/66128904/83275983-399bc200-a1d0-11ea-91cd-227e98791fb0.jpg)

Environment :
- Unity Version: Unity 2019.3.14f1
- OS + version: Windows 10
- ml-agents: 0.16.1,
- ml-agents-envs: 0.16.1,
- Communicator API: 1.0.0,
- TensorFlow: 2.2.0

Thank you very much, I hope you can help me !

**Comment by anupambhatnagar** [comment]
@Jean-Diddy The error message from tensorflow is harmless.

**Comment by Jean-Diddy** [comment]
> @anupambhatnagar Thank you. You think that without improving the performance of tensorflow by adding this AVX2 I should get the same results than displayed in the RollerBall tutorial ?

**Comment by Jean-Diddy** [comment]
I tried to decrease the time scale in the project settings and it gives me a better learning but it took a longer time to reach 1.00 of mean reward. I understood that the time scale can affect physic in ml-agents tool, and I wonder if there can be a huge difference in learning rate according to this Time scale and the CPU of the computer. Can the CPU limit a lot the learning if it is not so "powerful" ? Do the max time scale one can put without damaging the learning depends a lot on the CPU of one's computer ?

**Comment by DVonk** [comment]
I just noticed the exact same thing. I haven't used ML-Agents since 0.13 and it never took very long to learn RollerBall. But now I'm not even getting close to the performance shown in TensorBoard in the example. 

After 10k steps I was at a mean of 0.194. After 20k at 0.214. And it increased extremely slowly from then on. At 130k steps I was at 0.715, but then it went down again and hovered there for a very long time. At 240k it finally broke out to 0.794, but then down again. I'm at 300k steps right now with 0.785. I have no idea if it'll ever reach 1.

When is the last time you've tried this tutorial? Because it doesn't seem to work as well.

**Comment by Jean-Diddy** [comment]
Ok, that's what happened for me too ! I tried again yesterday and it worked but I do not understand why because I set the default parameters given in the RollerBall tutorial...I  don't understand, sometimes it works but most of the time it seems to fail...
I tried an other tutorial "Wall Jump" and my agent seems to learn pretty well how in the Small Jump configuration but stays as "dumb" as the begining without learning with the Big Wall configuration. The agent doesn't want to try jumping on the cube...
@DVonk did you try this tutorial and did you have the same issue ?
And about RollerBall, if you try over and over again, is there sometimes a great learning rate ?
